integration-tests/vertx-http-compressors/app/src/test/java/io/quarkus/compressors/it/Testflow.java

```diff
@@ -0,0 +1,120 @@
+package io.quarkus.compressors.it;
+
+import static io.netty.handler.codec.compression.ZlibCodecFactory.newZlibDecoder;
+import static io.netty.handler.codec.http.HttpResponseStatus.OK;
+import static io.quarkus.arc.ComponentsProvider.LOG;
+import static io.quarkus.compressors.it.CompressedResource.TEXT;
+import static java.lang.Integer.parseInt;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.embedded.EmbeddedChannel;
+import io.netty.handler.codec.compression.BrotliDecoder;
+import io.netty.handler.codec.compression.ZlibWrapper;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.ext.web.client.HttpResponse;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.client.WebClientOptions;
+
+public class Testflow {
+
+    // Depending on the defaults of the compression algorithm,
+    // the compressed content length may vary slightly between
+    // Vert.x/Netty versions over time.
+    public static final int COMPRESSION_TOLERANCE_BYTES = 2;
```

Comment:
2 bytes is quite small for a 5x factor compression. would 2% tolerance be better / less likely to break ?

Reply:
Yes, having a reasonable tolerance window would mitigate the potential issue and still be accurate enough.

Reply:
Implemented in [cc361d1](https://github.com/quarkusio/quarkus/pull/40750/commits/cc361d12fcd99dd7c021a4ab92384c1f560e93e0).